ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/LedgerDB.hs

```diff
@@ -1,82 +1,1987 @@
+{-# LANGUAGE BangPatterns               #-}
+{-# LANGUAGE CPP                        #-}
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingVia                #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GADTs                      #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE LambdaCase                 #-}
+{-# LANGUAGE NamedFieldPuns             #-}
+{-# LANGUAGE NumericUnderscores         #-}
+{-# LANGUAGE RankNTypes                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TupleSections              #-}
+{-# LANGUAGE TypeApplications           #-}
+{-# LANGUAGE TypeFamilies               #-}
+{-# LANGUAGE TypeOperators              #-}
+
+-- | The Ledger DB is responsible for the following tasks:
+--
+-- - __Maintaining the in-memory ledger state at the tip__: Maintaining the
+--     ledger state corresponding to the current tip in memory. When we try to
+--     extend our chain with a new block fitting onto our tip, the block must
+--     first be validated using the right ledger state, i.e., the ledger state
+--     corresponding to the tip. The current ledger state is needed for various
+--     other purposes.
+--
+-- - __Maintaining the past \(k\) in-memory ledger states__: we might roll back
+--     up to \(k\) blocks when switching to a more preferable fork. Consider the
+--     example below:
+--
+--     <<docs/haddocks/ledgerdb-switch.svg>>
```

Comment:
If I use just `make haddocks`, it seems that locally pictures and tikz code aren't displayed correctly. Example:

![image](https://github.com/input-output-hk/ouroboros-consensus/assets/47450644/404c3de7-1371-4c82-9983-61e6fba9ef59)

Reply:
It should be fixed now with the last commit